File: library/src/main/java/de/westnordost/osmfeatures/StringUtils.java
package de.westnordost.osmfeatures;

import java.text.Normalizer;
import java.util.Locale;
import java.util.regex.Pattern;

public class StringUtils
{
	private static final Pattern FIND_DIACRITICS = Pattern.compile("\\p{InCombiningDiacriticalMarks}+");

	public static String canonicalize(String str)
	{
		return stripDiacritics(str).toLowerCase(Locale.US);
	}

	private static String stripDiacritics(String str)
	{
		return FIND_DIACRITICS.matcher(Normalizer.normalize(str, Normalizer.Form.NFD)).replaceAll("");
	}
}

File: library/src/main/java/de/westnordost/osmfeatures/StringUtils.kt
package de.westnordost.osmfeatures

import java.util.regex.Pattern

import kotlin.text.Regex

class StringUtils {

    private val FIND_DIACRITICS = Pattern.compile("\\p{InCombiningDiacriticalMarks}+")


    companion object {
        @JvmStatic
        fun canonicalize(str: String): String {
            return stripDiacritics(str).lowercase()
        }

        private fun stripDiacritics(str: String): String {
            val reg = Regex("\\p{InCombiningDiacriticalMarks}+")
            return reg.replace("", str)
        }
    }




}